### PR TITLE
feat: Implement an endpoint to delete a note

### PR DESCRIPTION
**Pull request by ArzamastsevOleksandr**

### CODE REVIEW

File: src/main/java/com/booster/server/NoteController.java [hunk 1]
```diff
@@ -41,4 +41,10 @@ List<NoteDto> list(@RequestParam Integer size) {
         return noteService.batchOfSize(size);
     }
 
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
```
Comment (LarysaDoroshenko) [inline src/main/java/com/booster/server/NoteController.java:45]: Add the same response status for the update endpoint.

File: src/main/java/com/booster/server/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+public class GlobalExceptionHandler {
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+
```
Comment (LarysaDoroshenko) [inline src/main/java/com/booster/server/GlobalExceptionHandler.java:13]: Is this a redundant new line?

File: src/main/java/com/booster/server/NoteService.java [hunk 3]
```diff
@@ -42,4 +43,13 @@ public List<NoteDto> batchOfSize(Integer size) {
                 .toList();
     }
 
+    @Transactional
+    public void deleteById(Long id) {
+        try {
+            noteRepository.deleteById(id);
+        } catch (EmptyResultDataAccessException e) {
+            throw new EntityNotFoundException(NoteEntity.class, "id", id);
+        }
```
Comment (LarysaDoroshenko) [inline src/main/java/com/booster/server/NoteService.java:52]: EmptyResultDataAccessException has a quite good description. Why do we need to rethrow a  custom exception here?

File: src/test/java/com/booster/server/NoteControllerTest.java [hunk 4]
```diff
@@ -113,4 +113,38 @@ void returnsBatchOfNotesWithRespectToLastSeenAt() throws Exception {
...
+
+        assertThat(noteRepository.findById(id1)).isEmpty();
+        assertThat(noteRepository.findById(id2)).isNotEmpty();
+    }
+
+    @Test
+    void failsToDeleteById() throws Exception {
+        Long id1 = noteRepository.save(new NoteEntity()
+                        .setContent(firstNoteContent))
+                .getId();
```
Comment (LarysaDoroshenko) [inline src/test/java/com/booster/server/NoteControllerTest.java:137]: Do we need this successfully saved note if we test the fail scenario?

File: src/test/java/com/booster/server/NoteControllerTest.java [hunk 5]
```diff
@@ -113,4 +113,38 @@ void returnsBatchOfNotesWithRespectToLastSeenAt() throws Exception {
...
+
+        int missingId = 1000;
+
+        mockMvc.perform(MockMvcRequestBuilders.delete("/note/{id}", missingId))
+                .andDo(print())
+                .andExpect(status().isNotFound())
+                .andExpect(jsonPath("$.message").value("Entity not found [type=%s, notFoundBy=%s, attributeValue=%s]"
+                        .formatted(NoteEntity.class.getSimpleName(), "id", missingId)));
+
+        assertThat(noteRepository.findById(id1)).isNotEmpty();
```
Comment (LarysaDoroshenko) [inline src/test/java/com/booster/server/NoteControllerTest.java:147]: I think it is an unnecessary assertion as here we test exception throwing with a particular message.

File: src/main/java/com/booster/server/NoteController.java (hunk 1 above)
Comment (ArzamastsevOleksandr) [reply src/main/java/com/booster/server/NoteController.java:45]: Done

File: src/main/java/com/booster/server/GlobalExceptionHandler.java (hunk 2 above)
Comment (ArzamastsevOleksandr) [reply src/main/java/com/booster/server/GlobalExceptionHandler.java:13]: Fixed

File: src/main/java/com/booster/server/NoteService.java (hunk 3 above)
Comment (ArzamastsevOleksandr) [reply src/main/java/com/booster/server/NoteService.java:52]: `EmptyResultDataAccessException` is data access later specific and usually has no meaning to the end user.

Also, the message of the original exception is similar to `No class com.booster.server.NoteEntity entity with id 1000 exists!` which is a bit strange. We have no need of exposing the package structure to the end user, etc, so we need a custom exception.

File: src/test/java/com/booster/server/NoteControllerTest.java (hunk 4 above)
Comment (ArzamastsevOleksandr) [reply src/test/java/com/booster/server/NoteControllerTest.java:137]: That's a good question.

I added it to ensure that the existing notes are not affected by a failing call.

File: src/test/java/com/booster/server/NoteControllerTest.java (hunk 5 above)
Comment (ArzamastsevOleksandr) [reply src/test/java/com/booster/server/NoteControllerTest.java:147]: Agree, deleted the redundant stuff.

File: src/main/java/com/booster/server/NoteService.java (hunk 3 above)
Comment (LarysaDoroshenko) [reply src/main/java/com/booster/server/NoteService.java:52]: Okay

**Review by LarysaDoroshenko — APPROVED** [review]

**Pull request merged**